<?xml version="1.0"?>

<launch>
  <arg name="odomTopic" default="/gmapping/odometry" />
	<arg name="cmdTopic" default="/mpc_vel" />
  <!--<arg name="param_file" value="meta_mpc/zz_double_integrator_mpcc0" />-->
  <arg name="param_file" value="zz_unicycle_model_mpcc1" />
  <arg name="use_rl" default="false" />
  <arg name="db_filename" default="$(env HOME)/dbs/rl_learning.db" />

  <rosparam file="$(find mpcc)/params/meta_mpc/$(arg param_file).yaml" command="load" ns="mpcc/" />
  <rosparam file="$(find mpcc)/params/robo_params.yaml" command="load" ns="mpcc/" />

  <node name="pearl_server" pkg="mpcc" type="pearl_server.py" output="screen">
    <param name="param_file" value="$(arg param_file)" />
  </node>

  <!--<node name="mpcc" pkg="mpcc" type="mpcc_ros" launch-prefix="lldb --><!--" output="screen"> -->
	<node name="mpcc" pkg="mpcc" type="mpcc_ros" output="screen">
		<!--<param name="mpc_type" value="$(arg mpc_type)" />-->

		<remap from="/odometry/filtered" to="$(arg odomTopic)" />
		<!--<remap from="/cmd_vel" to="$(arg cmdTopic)" />-->
	</node>

	<rosparam file="$(find mpcc)/params/local_costmap_params.yaml" command="load" ns="local_costmap/local_costmap/" />
	<node name="local_costmap" pkg="mpcc" type="local_costmap" output="screen" />
	<!--<node name="cmd_mapper" pkg="mpcc" type="di_to_uni_cmd_mapper.py" output="screen" />-->

  <include file="$(find mpcc)/launch/gmapping.launch" /> 

</launch>

